Add FramebufferCreateInfoBuilder::attachment_count

Currently, it only has `attachments`, which require the actual slice to be passed in.
However, in the image-less framebuffer scenario, only the count needs to be set. So the builder isn't compatible with this right now.

> The suggestion to do `&vec![vk::ImageView::null(); runtime_value]` is very dangerous since it breaks if one calls `build()` in unexpected ways.

It is _exactly_ the same as borroing the **temporary** array (unless making it `static` or `const`):

```rust
error[E0716]: temporary value dropped while borrowed
  --> examples/src/bin/texture.rs:91:35
   |
91 |                     .attachments(&[vk::ImageView::null(); 3])
   |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^ creates a temporary which is freed while still in use
92 |                     .layers(1);
   |                               - temporary value is freed at the end of this statement
...
95 |                     .create_framebuffer(&frame_buffer_create_info, None)
   |                                         ------------------------- borrow later used here
   |
   = note: consider using a `let` binding to create a longer lived value
```

Both are useless except if creating the object/builder within the same _expression_ as `create_framebuffer`.

> I suppose you just don't want to make an exception for this

I very well want to make an exception for this **_as mentioned above_**, I just don't want to make it the default for other builders where slice-length and slice-pointer are completely tied together. Regardless of general unsafety, Ash builders try to hold ones hand to the best of our abilities when pointer-lifetime is concerned.

--- a/ash/src/vk/definitions.rs
+++ b/ash/src/vk/definitions.rs
@@ -7276,6 +7276,11 @@ impl<'a> FramebufferCreateInfo<'a> {
         self
     }
     #[inline]
+    pub fn attachment_count(mut self, attachment_count: u32) -> Self {
+        self.attachment_count = attachment_count;
+        self
+    }
+    #[inline]
     pub fn attachments(mut self, attachments: &'a [ImageView]) -> Self {
         self.attachment_count = attachments.len() as _;
         self.p_attachments = attachments.as_ptr();

--- a/generator/src/lib.rs
+++ b/generator/src/lib.rs
@@ -1786,10 +1786,15 @@ pub fn derive_setters(
     // Must either have both, or none:
     assert_eq!(next_field.is_some(), structure_type_field.is_some());
 
-    let nofilter_count_members = [
-        ("VkPipelineViewportStateCreateInfo", "pViewports"),
-        ("VkPipelineViewportStateCreateInfo", "pScissors"),
-        ("VkDescriptorSetLayoutBinding", "pImmutableSamplers"),
+    let allowed_count_members = [
+        // pViewports is allowed to be empty if the viewport state is empty
+        ("VkPipelineViewportStateCreateInfo", "viewportCount"),
+        // Must match viewportCount
+        ("VkPipelineViewportStateCreateInfo", "scissorCount"),
+        // descriptorCount is settable regardless of having pImmutableSamplers
+        ("VkDescriptorSetLayoutBinding", "descriptorCount"),
+        // No ImageView attachments when VK_FRAMEBUFFER_CREATE_IMAGELESS_BIT is set
+        ("VkFramebufferCreateInfo", "attachmentCount"),
     ];
     let filter_members: Vec<String> = members
         .iter()
@@ -1799,7 +1804,7 @@ pub fn derive_setters(
             // Associated _count members
             if field.array.is_some() {
                 if let Some(ref array_size) = field.size {
-                    if !nofilter_count_members.contains(&(&struct_.name, field_name)) {
+                    if !allowed_count_members.contains(&(&struct_.name, array_size)) {
                         return Some((*array_size).clone());
                     }
                 }

--- a/generator/src/lib.rs
+++ b/generator/src/lib.rs
@@ -1796,28 +1796,21 @@ pub fn derive_setters(
         // No ImageView attachments when VK_FRAMEBUFFER_CREATE_IMAGELESS_BIT is set
         ("VkFramebufferCreateInfo", "attachmentCount"),
     ];
-    let filter_members: Vec<String> = members
+    let filter_members = members
         .iter()
         .filter_map(|(field, _)| {
-            let field_name = field.name.as_ref().unwrap();
-
             // Associated _count members
             if field.array.is_some() {
-                if let Some(ref array_size) = field.size {
+                if let Some(array_size) = &field.size {
                     if !allowed_count_members.contains(&(&struct_.name, array_size)) {
-                        return Some((*array_size).clone());
+                        return Some(array_size);
                     }
                 }
             }
 
-            // VkShaderModuleCreateInfo requires a custom setter
-            if field_name == "codeSize" {
-                return Some(field_name.clone());
-            }
-
             None
         })
-        .collect();
+        .collect::<Vec<_>>();
 
     let setters = members.iter().filter_map(|(field, deprecated)| {
         let deprecated = deprecated.as_ref().map(|d| quote!(#d #[allow(deprecated)]));
@@ -1836,12 +1829,15 @@ pub fn derive_setters(
         let mut param_ident_short = format_ident!("{}", param_ident_short);
 
         if let Some(name) = field.name.as_ref() {
-            // Filter
-            if filter_members.iter().any(|n| *n == *name) {
+            if filter_members.contains(&name) {
                 return None;
             }
 
             // Unique cases
+            if struct_.name == "VkShaderModuleCreateInfo" && name == "codeSize" {
+                return None;
+            }
+
             if struct_.name == "VkShaderModuleCreateInfo" && name == "pCode" {
                 return Some(quote!{
                     #[inline]

